cli/dts/lib.deno.unstable.d.ts

```diff
@@ -117,16 +117,59 @@ declare namespace Deno {
     | "usize"
     | "isize"
     | "f32"
-    | "f64";
+    | "f64"
+    | "buffer";
 
   /** A foreign function as defined by its parameter and result types */
   export interface ForeignFunction {
-    parameters: (NativeType | "buffer")[];
+    parameters: NativeType[];
     result: NativeType;
     /** When true, function calls will run on a dedicated blocking thread and will return a Promise resolving to the `result`. */
     nonblocking?: boolean;
   }
 
+  type TypedArray =
+    | Int8Array
+    | Uint8Array
+    | Int16Array
+    | Uint16Array
+    | Int32Array
+    | Uint32Array
+    | Uint8ClampedArray
+    | Float32Array
+    | Float64Array
+    | BigInt64Array
+    | BigUint64Array;
+
+  /** An unsafe pointer to an memory location for passing and returning pointers to and from the ffi */
+  export class UnsafePointer {
+    constructor(value: bigint);
+
+    value: bigint;
+
+    /**
+     * Return the direct memory pointer to the typed array in memory
+     */
+    static of(typedArray: TypedArray): UnsafePointer;
+
+    /**
+     * Reads memory at the pointer into a typed array.
+     * Length is determined from the typed array's `byteLength`.
+     * Also takes optional offset.
+     */
+    read(destination: TypedArray, offset?: bigint | number): void;
```

Comment:
My understanding is that this would copy data, would be nice to have an view based approach over a said pointer into native memory that uses https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DataView as a way to read typed information from the raw memory the pointer points to.

Reply:
Perhaps a read only (kinda dangerous writing to arbitrary pointers, but maybe...) copy of the `DataView` API? Although there is a few questions i have with it:
1. It would have to work with `bigint` offsets instead of `numbers`
2. Why would you want to create more than one instance of this new `DataView` copy as they all address the same memory? Or should it be an extension to the `UnsafePointer` API?
3. The `read` method should probably also be renamed `copy` as that more accurately reflects what it does, or maybe we should even add a `getArrayBuffer(byteLength: bigint | number, offset?: bigint | number): ArrayBuffer` (or `Uint8Array` maybe?)

Reply:
The new structure should mimic the `DataView` API, work with `bigint` as you said.

I think having 2 methods on the `UnsafePointer` will be nice:
- `UnsafePointer::toReadonlyView(offset:bignum, size: bignum): ReadOnlyPointerView`
- `UnsafePointer::toWritableView(offset:bignum, size: bignum): WritablePointerView`
Where `WritablePointerView` would extend `ReadOnlyPointerView` and allow write access to the underlying memory. This is just a proposal so naming and form could be definitely be improved...

I agree that the `read` method should be named to `copy` as it as clear as possible about the intent.
Maybe create a new PR with this new `View` APIs so it is easier to review and make this one mergeable quicker?

Reply:
> The `read` method should probably also be renamed `copy` as that more accurately reflects what it does, or maybe we should even add a `getArrayBuffer(byteLength: bigint | number, offset?: bigint | number): ArrayBuffer` (or `Uint8Array` maybe?)

I had it to be named `read` in the proposal as it seemed consistent with `Deno.Reader#read`. But rename to `copy` would make sense, I agree.

Reply:
> Maybe create a new PR with this new `View` APIs so it is easier to review and make this one mergeable quicker?

Ideally yes, but should this PR then leave out the `read` and `readCString` methods in favour of that future pr?